### PR TITLE
Update CommandLibrary to CommandRequester

### PR DESCRIPTION
**Pull request by DanielCuestaJibo**
https://jira.jibo.com/browse/JIBO-10445

### CODE REVIEW

File: src/main/java/com/jibo/apptoolkit/protocol/CommandRequester.java [hunk 1]
```diff
@@ -623,4 +487,329 @@ public void onListen(String transactionID, String speech) {}
...
+
+    abstract static class Base {
+        CommandRequester commandRequester;
+
+        Base(CommandRequester commandRequester) {
+            this.commandRequester = commandRequester;
+        }
+    }
+
+    public static class Assets extends Base {
```
Comment (susanjibo) [inline src/main/java/com/jibo/apptoolkit/protocol/CommandRequester.java]: needs docs

File: src/main/java/com/jibo/apptoolkit/protocol/CommandRequester.java [hunk 2]
```diff
@@ -623,4 +487,329 @@ public void onListen(String transactionID, String speech) {}
...
+         * @param uri URI to the asset to be fetched
+         * @param name Name the asset will be called by
+         * @param onCommandResponseListener {@link OnCommandResponseListener#onEvent(String, EventMessage.BaseEvent)}
+         */
+        public String load(String uri, String name, OnCommandResponseListener onCommandResponseListener) {
+            return commandRequester.sendCommand(new Command.FetchAssetRequest(uri, name), onCommandResponseListener);
+        }
+    }
+
+    public static class Display extends Base{
```
Comment (susanjibo) [inline src/main/java/com/jibo/apptoolkit/protocol/CommandRequester.java]: needs docs

File: src/main/java/com/jibo/apptoolkit/protocol/CommandRequester.java [hunk 3]
```diff
@@ -623,4 +487,329 @@ public void onListen(String transactionID, String speech) {}
...
+
+        Subscribe subscribe;
+
+        Display(CommandRequester commandRequester) {
+            super(commandRequester);
+            subscribe = new Subscribe(commandRequester);
+        }
+
+        public Subscribe getSubscribe() {
```
Comment (susanjibo) [inline src/main/java/com/jibo/apptoolkit/protocol/CommandRequester.java]: what is this? docs

File: src/main/java/com/jibo/apptoolkit/protocol/CommandRequester.java [hunk 4]
```diff
@@ -623,4 +487,329 @@ public void onListen(String transactionID, String speech) {}
...
+        /**
+         * Display something on Jibo's screen
+         * @param view What to display onscreen. See Command.DisplayRequest.DisplayView
+         * @param onCommandResponseListener {@link OnCommandResponseListener#onEvent(String, EventMessage.BaseEvent)}
+         */
+        String display(Command.DisplayRequest.DisplayView view, OnCommandResponseListener onCommandResponseListener) {
+            return commandRequester.sendCommand(new Command.DisplayRequest(view), onCommandResponseListener);
+        }
+
+        public String text(Command.DisplayRequest.TextView view, OnCommandResponseListener onCommandResponseListener) {
```
Comment (susanjibo) [inline src/main/java/com/jibo/apptoolkit/protocol/CommandRequester.java]: docs

File: src/main/java/com/jibo/apptoolkit/protocol/CommandRequester.java [hunk 5]
```diff
@@ -623,4 +487,329 @@ public void onListen(String transactionID, String speech) {}
...
+            return display(view, onCommandResponseListener);
+        }
+
+        public String image(Command.DisplayRequest.ImageView view, OnCommandResponseListener onCommandResponseListener) {
```
Comment (susanjibo) [inline src/main/java/com/jibo/apptoolkit/protocol/CommandRequester.java]: docs

File: src/main/java/com/jibo/apptoolkit/protocol/CommandRequester.java [hunk 6]
```diff
@@ -623,4 +487,329 @@ public void onListen(String transactionID, String speech) {}
...
+            return display(view, onCommandResponseListener);
+        }
+
+        public String eye(Command.DisplayRequest.EyeView view, OnCommandResponseListener onCommandResponseListener) {
```
Comment (susanjibo) [inline src/main/java/com/jibo/apptoolkit/protocol/CommandRequester.java]: docs

File: src/main/java/com/jibo/apptoolkit/protocol/CommandRequester.java [hunk 7]
```diff
@@ -623,4 +487,329 @@ public void onListen(String transactionID, String speech) {}
...
+             */
+            public String gesture(Command.ScreenGestureRequest.ScreenGestureFilter filter, OnCommandResponseListener onCommandResponseListener) {
+                commandRequester.closeGestureConnection();
+
+                return commandRequester.sendCommand(new Command.ScreenGestureRequest(filter), onCommandResponseListener);
+            }
+        }
+    }
+
+    public static class Perception {
```
Comment (susanjibo) [inline src/main/java/com/jibo/apptoolkit/protocol/CommandRequester.java]: docs

File: src/main/java/com/jibo/apptoolkit/protocol/CommandRequester.java [hunk 8]
```diff
@@ -623,4 +487,329 @@ public void onListen(String transactionID, String speech) {}
...
+        public static class Subscribe extends Base {
+
+            Subscribe(CommandRequester commandRequester) {
+                super(commandRequester);
+            }
+
+            /**
+             * Get face entity
+             * @param onCommandResponseListener {@link OnCommandResponseListener#onEvent(String, EventMessage.BaseEvent)}
+             * @return
```
Comment (susanjibo) [inline src/main/java/com/jibo/apptoolkit/protocol/CommandRequester.java]: @return what?

File: src/main/java/com/jibo/apptoolkit/protocol/CommandRequester.java [hunk 9]
```diff
@@ -623,4 +487,329 @@ public void onListen(String transactionID, String speech) {}
...
+             * @param onCommandResponseListener {@link OnCommandResponseListener#onEvent(String, EventMessage.BaseEvent)}
+             */
+            public String motion(OnCommandResponseListener onCommandResponseListener){
+                return commandRequester.sendCommand(new Command.MotionRequest(), onCommandResponseListener);
+            }
+
+            /**
+             * Listen for head touch
+             * @param onCommandResponseListener {@link OnCommandResponseListener#onEvent(String, EventMessage.BaseEvent)}
+             * @return
```
Comment (susanjibo) [inline src/main/java/com/jibo/apptoolkit/protocol/CommandRequester.java]: @return what?

File: src/main/java/com/jibo/apptoolkit/protocol/CommandRequester.java [hunk 10]
```diff
@@ -623,4 +487,329 @@ public void onListen(String transactionID, String speech) {}
...
+             */
+            public String headTouch(OnCommandResponseListener onCommandResponseListener){
+                return commandRequester.sendCommand(new Command.HeadTouchRequest(), onCommandResponseListener);
+            }
+        }
+
+    }
+
+    public static class Config extends Base {
```
Comment (susanjibo) [inline src/main/java/com/jibo/apptoolkit/protocol/CommandRequester.java]: docs

File: src/main/java/com/jibo/apptoolkit/protocol/CommandRequester.java [hunk 11]
```diff
@@ -623,4 +487,329 @@ public void onListen(String transactionID, String speech) {}
...
+
+        Config(CommandRequester commandRequester) {
+            super(commandRequester);
+        }
+
+        /**
+         * Set robot configuration data.
+         * @param options Options to set. See Command.SetConfigRequest.SetConfigOptions.
+         * @param onCommandResponseListener {@link OnCommandResponseListener#onEvent(String, EventMessage.BaseEvent)}
+         * @return
```
Comment (susanjibo) [inline src/main/java/com/jibo/apptoolkit/protocol/CommandRequester.java]: @return what

File: src/main/java/com/jibo/apptoolkit/protocol/CommandRequester.java [hunk 12]
```diff
@@ -623,4 +487,329 @@ public void onListen(String transactionID, String speech) {}
...
+         */
+        public String set(Command.SetConfigRequest.SetConfigOptions options, OnCommandResponseListener onCommandResponseListener){
+            return commandRequester.sendCommand(new Command.SetConfigRequest(options), onCommandResponseListener);
+        }
+
+        /**
+         * Get robot configuration data
+         * @param onCommandResponseListener {@link OnCommandResponseListener#onEvent(String, EventMessage.BaseEvent)}
+         * @return
```
Comment (susanjibo) [inline src/main/java/com/jibo/apptoolkit/protocol/CommandRequester.java]: @return what

File: src/main/java/com/jibo/apptoolkit/protocol/CommandRequester.java [hunk 13]
```diff
@@ -623,4 +487,329 @@ public void onListen(String transactionID, String speech) {}
...
+         */
+        public String get(OnCommandResponseListener onCommandResponseListener){
+            return commandRequester.sendCommand(new Command.GetConfigRequest(), onCommandResponseListener);
+        }
+    }
+
+    public static class Listen extends Base {
```
Comment (susanjibo) [inline src/main/java/com/jibo/apptoolkit/protocol/CommandRequester.java]: docs

File: src/main/java/com/jibo/apptoolkit/protocol/CommandRequester.java [hunk 14]
```diff
@@ -623,4 +487,329 @@ public void onListen(String transactionID, String speech) {}
...
+         *             See <a href="https://app-toolkit.jibo.com/esml/">App Toolkit Docs</a> for ESML info.
+         * @param onCommandResponseListener {@link OnCommandResponseListener#onEvent(String, EventMessage.BaseEvent)}
+         */
+        public String say(String text, OnCommandResponseListener onCommandResponseListener) {
+            if (StringUtils.isEmpty(text)) return null;
+            return commandRequester.sendCommand(new Command.SayRequest(text), onCommandResponseListener);
+        }
+    }
+
+    public static class Media extends Base {
```
Comment (susanjibo) [inline src/main/java/com/jibo/apptoolkit/protocol/CommandRequester.java]: docs

File: src/main/java/com/jibo/apptoolkit/protocol/CommandRequester.java [hunk 15]
```diff
@@ -623,4 +487,329 @@ public void onListen(String transactionID, String speech) {}
...
+                commandRequester.closeVideoConnection();
+
+                if (videoType == null) return null;
+                return commandRequester.sendCommand(duration > 0 ? new Command.VideoRequest(videoType, duration)
+                        : new Command.VideoRequest(videoType), onCommandResponseListener);
+            }
+        }
+    }
+
+    public static class Session extends Base {
```
Comment (susanjibo) [inline src/main/java/com/jibo/apptoolkit/protocol/CommandRequester.java]: docs

File: src/main/java/com/jibo/apptoolkit/protocol/CommandRequester.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (DanielCuestaJibo) [reply src/main/java/com/jibo/apptoolkit/protocol/CommandRequester.java]: @calvin-j-park @susanjibo 

We have the next methods in javascript:
display.createEyeView
display.createImageView
display.createTextView
display.swapView

and the united methods in java + swift should be:
display.swap
display.<verb>

I understand that createEyeView, createImageView and createTextView instructions should be display.<verb>:
display.eye
display.image
display.text

and swapView should be display.swap

Is that correct? or should we use display.swap to make eye, image and text instructions?

**Comment by susanjibo** [comment]
Can we include `mividtim` on this repo so Tim can make the necessary changes to the java desktop example? thanks

**Comment by mbittarelli-jibo** [comment]
@susanjibo @mividtim 
done

**Comment by susanjibo** [comment]
What is this PR waiting on?

**Pull request merged**